### PR TITLE
Add debug configuration

### PR DESCRIPTION
**Pull request by mos9527**
TL;DR: This PR address the issue where the Windows MSVC builds of the extension would cause a `Read Access Violation` crash due to mismatched Visual C++ Redistributable DLLs on certain platforms.

Additionally, this PR adds an environment build flag (`DEBUG`) for MSVC builds to allow the extension to be built with debug information, since `pip` doesn't seem to provide a switch for it. This is used in my own debugging process.

A (not so formal) write up about the change is posted here, detailing one edge case where this issue may actually manifest.
- https://github.com/mos9527/sssekai_blender_io/issues/11

The fix proposed by this PR does seem to remedy the said issue as per my limited testing.

### CODE REVIEW

**Comment by K0lb3** [comment]
Thanks for the write-up and report.

As vcruntime itself is a dependency that might not be available the better solution is to statically link it using `` "/MT"``, or alternatively fixing the wheel using delvewheel which would simply include the required dlls.
Given the span of covered simd options the later option is better size-wise.
I will make a fix for that later in the workflow.

I would still like to keep your debug part.
It would be nice if you could extend it for gcc/clang so that the feature is cross-platform.

**Comment by mos9527** [comment]
Compiler flags `-O0 -ggdb` seem to work for gcc. [I added it to non-MSVC compiler flags with the environ enabled](https://github.com/K0lb3/astc-encoder-py/pull/2/commits/aa89bba13a1cf9dd931dececb5f584cd19119811).
```_encoder_none.abi3.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=6440cea390fb70a42acfa95aa1714a8e61fe71e5, with debug_info, not stripped```

For accuracy I've changed the environ name from `DEBUG` to `DEBUG_INFO` since the flags only disabled optimization & enabled debug information to be produced. 
Which is different from builds with `build_ext.debug` flag enabled - the latter links with debug runtime libaries (e.g. `python3xx_d.dll`, `msvcpxxxd.dll`, etc on Windows). More info available [here](https://github.com/pypa/setuptools/blob/main/setuptools/command/build_ext.py) and [here](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/_msvccompiler.py).

> As vcruntime itself is a dependency that might not be available the better solution is to statically link it using  "/MT", or alternatively fixing the wheel using delvewheel which would simply include the required dlls.

I've tried `delvewheel` locally and it does seem to work. Thank you in advance for the fix!

**Review by K0lb3 — APPROVED** [review]
